openedx_external_enrollments/external_enrollments/icc_external_enrollment.py

```diff
@@ -0,0 +1,198 @@
+"""ICCExternalEnrollment class file."""
+import json
+import logging
+from uuid import uuid4
+
+from django.conf import settings
+from rest_framework import status
+import requests
+import xmltodict
+
+from openedx_external_enrollments.external_enrollments.base_external_enrollment import BaseExternalEnrollment
+from openedx_external_enrollments.edxapp_wrapper.get_student import get_user
+from openedx_external_enrollments.models import EnrollmentRequestLog
+
+LOG = logging.getLogger(__name__)
+
+
+class ICCExternalEnrollment(BaseExternalEnrollment):
+    """
+    ICCExternalEnrollment class.
+    """
+    USER_CREATION_MAX_RETRIES = 5
+    USERNAME_SUFFIX_LENGTH = 5
+
+    def __str__(self):
+        return "ICC"
+
+    def _execute_post(self, url, data=None, headers=None, json_data=None):
+        """
+        Execute post request to achieve the ICC external enrollment.
+        """
+        response = requests.post(
+            url=url,
+            data=json_data,
+        )
+
+        return response
+
+    def _get_enrollment_headers(self):
+        """
+        Method that returns None by default, ICC integration does not headers.
+        """
+        return None
+
+    def _get_enrollment_data(self, data, course_settings):
+        """
+        Method that provide the relevant data for the enrollment.
+        """
+
+        enrollment_data = {}
+        try:
+            icc_user = self._get_icc_user(data)
+            enrollment_data = {
+                "wstoken": settings.ICC_API_TOKEN,
+                "wsfunction": settings.ICC_ENROLLMENT_API_FUNCTION,
+                "enrolments[0][roleid]": settings.ICC_LEARNER_ROLE_ID,
+                "enrolments[0][userid]": icc_user.get('id'),
+                "enrolments[0][courseid]": course_settings.get("external_course_run_id"),
+            }
+        except Exception:
+            pass
+
+        return enrollment_data
+
+    def _get_enrollment_url(self, course_settings):
+        """
+        Method that returns base url for the ICC integration API.
+        """
+
+        return settings.ICC_BASE_URL
+
+    def _get_icc_user(self, data):
+        """
+        Method that look for the user in ICC database through the API, if the user is not
+        found then it calls create method.
+        """
+        log_details = {
+            "url": settings.ICC_API_TOKEN,
+        }
+        icc_user = {}
+
+        try:
+            _data = {
```

Comment:
What is this underscore for? @ivanvgh This variable can be called: request_data or so.

Reply:
'_data' renamed to 'request_data'.